src/components/structures/RoomDirectory.js

```diff
@@ -114,11 +116,21 @@ module.exports = React.createClass({
             };
         }
 
-        dis.dispatch({
-            action: 'view_room',
-            room_id: roomId,
+        var payload = {
             oob_data: oob_data,
-        });
+        };
+        if (roomAlias) {
+            payload.action = 'view_room_alias';
+            payload.room_alias = roomAlias;
```

Comment:
Is it a mistake to throw away the room id if we know it? I'm worried about cases where the directory somehow gets out of sync with the `room_aliases` in the rooms, so you actually end up joining a different room to the one you chose in the directory.

Perhaps at least a comment to this effect?